### PR TITLE
ODBC Based Databricks Connector

### PR DESCRIPTION
**Pull request by arikfr**
## What type of PR is this? 

- [x] Refactor

## Description

Updated the Databricks connector to use their ODBC driver instead of the PyHive connector.

### CODE REVIEW

File: client/app/components/CreateSourceDialog.jsx [hunk 1]
```diff
@@ -116,6 +116,15 @@ class CreateSourceDialog extends React.Component {
           )}
         </div>
         <DynamicForm id="sourceForm" fields={fields} onSubmit={this.createSource} feedbackIcons hideSubmitButton />
+        {selectedType.type === "databricks" && (
+          <small>
+            By using the Databricks Data Source you agree to the Databricks JDBC/ODBC{" "}
+            <a href="https://databricks.com/spark/odbc-driver-download" target="_blank" rel="noopener noreferrer">
```
Comment (susodapop) [inline client/app/components/CreateSourceDialog.jsx:122]: I don't see the terms and conditions on this page.

driver-download-page

File: redash/query_runner/databricks.py [hunk 2]
```diff
@@ -1,17 +1,50 @@
...
+
+def _build_odbc_connection_string(**kwargs):
+    connection_string = ""
+    for k, v in kwargs.items():
+        if connection_string:
+            connection_string = "{};{}={}".format(connection_string, k, v)
+        else:
+            connection_string = "{}={}".format(k, v)
+
+    return connection_string
```
Comment (susodapop) [inline redash/query_runner/databricks.py:41]: Immutable version.

```suggestion
def _build_odbc_connection_string(**kwargs):
    return ";".join([f"{k}={v}" for k, v in kwargs.items()])
```

File: redash/query_runner/databricks.py [hunk 3]
```diff
@@ -26,67 +59,98 @@ def configuration_schema(cls):
...
+            "order": ["host", "http_path", "http_password"],
             "secret": ["http_password"],
-            "required": ["host", "database", "http_path", "http_password"],
+            "required": ["host", "http_path", "http_password"],
         }
 
-    def _get_connection(self):
-        host = self.configuration["host"]
+    def _get_cursor(self):
+        connection_string = _build_odbc_connection_string(
```
Comment (susodapop) [inline redash/query_runner/databricks.py:73]: Worth documenting where this came from in the [DB Documentation](https://docs.databricks.com/integrations/bi/jdbc-odbc-bi.html#odbc-data-source-name-dsn-configuration-for-the-simba-odbc-driver)?

File: client/app/components/CreateSourceDialog.jsx (hunk 1 above)
Comment (arikfr) [reply client/app/components/CreateSourceDialog.jsx:122]: I can see it. Maybe some adblocker?

File: redash/query_runner/databricks.py (hunk 2 above)
Comment (arikfr) [reply redash/query_runner/databricks.py]: 👍 thanks.

File: redash/query_runner/databricks.py (hunk 3 above)
Comment (arikfr) [reply redash/query_runner/databricks.py:66]: It's not from there :) It's from the tableau connector implementation.

File: client/app/components/CreateSourceDialog.jsx (hunk 1 above)
Comment (susodapop) [reply client/app/components/CreateSourceDialog.jsx:122]: Strange. It's not an adblocker. Just not compatible in Firefox. Loads fine in Chrome and Safari.

**Pull request merged**